src/StdChains.sol

```diff
@@ -20,71 +22,126 @@ abstract contract StdChains {
+            bytes(foundAlias).length == 0 || keccak256(bytes(foundAlias)) == keccak256(bytes(chainAlias)),
             string(
                 abi.encodePacked(
-                    "StdChains setChain(string,string,uint256,string): Chain ID ",
-                    vm.toString(chainId),
+                    "StdChains setChain(string,Chain): Chain ID ",
+                    vm.toString(chain.chainId),
                     " already used by \"",
-                    idToKey[chainId],
+                    foundAlias,
                     "\"."
                 )
             )
         );
 
-        uint256 oldChainId = chains[key].chainId;
-        delete idToKey[oldChainId];
+        uint256 oldChainId = _chains[chainAlias].chainId;
+        delete _idToAlias[oldChainId];
 
-        chains[key] = Chain(name, chainId, rpcUrl);
-        idToKey[chainId] = key;
+        _chains[chainAlias] = chain;
+        _idToAlias[chain.chainId] = chainAlias;
     }
 
     function initialize() private {
         if (initialized) return;
-
-        setChain("anvil", "Anvil", 31337, "http://127.0.0.1:8545");
-        setChain("mainnet", "Mainnet", 1, "https://mainnet.infura.io/v3/6770454bc6ea42c58aac12978531b93f");
-        setChain("goerli", "Goerli", 5, "https://goerli.infura.io/v3/6770454bc6ea42c58aac12978531b93f");
-        setChain("sepolia", "Sepolia", 11155111, "https://rpc.sepolia.dev");
-        setChain("optimism", "Optimism", 10, "https://mainnet.optimism.io");
-        setChain("optimism_goerli", "Optimism Goerli", 420, "https://goerli.optimism.io");
-        setChain("arbitrum_one", "Arbitrum One", 42161, "https://arb1.arbitrum.io/rpc");
-        setChain("arbitrum_one_goerli", "Arbitrum One Goerli", 421613, "https://goerli-rollup.arbitrum.io/rpc");
-        setChain("arbitrum_nova", "Arbitrum Nova", 42170, "https://nova.arbitrum.io/rpc");
-        setChain("polygon", "Polygon", 137, "https://polygon-rpc.com");
-        setChain("polygon_mumbai", "Polygon Mumbai", 80001, "https://rpc-mumbai.matic.today");
-        setChain("avalanche", "Avalanche", 43114, "https://api.avax.network/ext/bc/C/rpc");
-        setChain("avalanche_fuji", "Avalanche Fuji", 43113, "https://api.avax-test.network/ext/bc/C/rpc");
-        setChain("bnb_smart_chain", "BNB Smart Chain", 56, "https://bsc-dataseed1.binance.org");
-        setChain("bnb_smart_chain_testnet", "BNB Smart Chain Testnet", 97, "https://data-seed-prebsc-1-s1.binance.org:8545");// forgefmt: disable-line
-        setChain("gnosis_chain", "Gnosis Chain", 100, "https://rpc.gnosischain.com");
-
-        // Loop over RPC URLs in the config file to replace the default RPC URLs
-        Vm.Rpc[] memory rpcs = vm.rpcUrlStructs();
-        for (uint256 i = 0; i < rpcs.length; i++) {
-            chains[rpcs[i].key].rpcUrl = rpcs[i].url;
-        }
-
         initialized = true;
+
+        setChainWithDefaultRpcUrl("anvil", Chain("Anvil", 31337, "http://127.0.0.1:8545"));
```

Comment:
@adhusson @ZeroEkkusu can we get some code comments in this PR, and maybe after some book updates, for StdChains / this PR? There's now a lot going on in `StdChains` and it's fairly complex, so the UX isn't the most intuitive by skimming the code (and there's a lot to skim). To start, a natspec block right above the `abstract contract StdChains {` line would be helpful. Some questions I'd like answered by those comments:
1. What's the default, out of the box behavior when I:
    1. Access a chain with a pre-existing alias, like "mainnet"
    2. Access a chain without a pre-existing alias, like "myNewChain"
    3.  For (1) and (2) how does it vary if nothing is in my foundry.toml vs. if I have a hardcoded RPC URL vs. if I have an env var, and that env var does or does not exist
2. How do I add a new chain?
3. What's the difference between `setChain` and `setChainWithDefaultRpcUrl` and when do I use each?
4. How do I rename an existing chain, like changing `anvil` to `hardhat`?

Reply:
Btw this function should also be only for devs and have the same discouraging naming style.

Reply:
There are 3 functions: 2 `getChain` and 1 `setChain`. The first time any one of these is called, chains are initialized with the default set of RPC URLs. This is done by `initialize`, which uses `setChainWithDefaultRpcUrl`. Defaults are recorded in `defaultRpcUrls`.

`setChain` is straightforward. It just sets a chain (in `chains`).

`getChain` uses `withRpcUrl` to return a chain. For example, let's say we want to retrieve `mainnet`. If you haven't set it with `setChain` and you haven't specified it in `foundry.toml`, the default one will be returned. If you set it in `foundry.toml` it will return that, if valid (e.g. the env exists), otherwise, the default one. If you specified it with `setChain` it will return that.

Reply:
Thanks for the docs, I modified them slightly and ported them as comments in 4b9b197. We can pull that and reuse it for the book PR too